### PR TITLE
Qs-Service-Agreements

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#16836

*NOTE:*
- Due to some pending design decision, i've commented out the document preview for now. 
- This means the above ticket will probably remain in staging until a decision and update is implemented, but the rest of the code can go in functionally with no issue.

*Description of changes:*
- Implements GET on Service Agreement document (future will consider versions, right now just the latest)
- Misc style and review section updates 
- Update validations for User Access flow
- Checks for Products in Conjunction with Roles for MHR Tabbed Table (Now that products are in place)
- Created jest setup file to mock out pdf embed - can be used to run any code before any/all jest unit tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1499-k6pag3qz.web.app

**Comment by dimak1** [comment]
@cameron-eyds do you want this reviewed or still wip?

**Comment by cameron-eyds** [comment]
> @cameron-eyds do you want this reviewed or still wip?

There was some unmentioned requirements discovered yesterday with regards to the preview of the Service Agreement. 
The requirement is now that we re-write the document into HTML to display vs having a pre-viewable or embedded pdf.

So that is now underway, this PR will probably be updated tomorrow, will post for review.
Thanks for checking sir!

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1499-k6pag3qz.web.app

**Review by doug-lovett — APPROVED** [review]
Stupendous!

File: ppr-ui/src/components/userAccess/ReviewConfirm/QsInformationReview.vue [hunk 1]
```diff
@@ -1,32 +1,44 @@
...
+    </template>
+
     <!-- Conditional based on service agreement checkbox -->
-    <template v-slot:topInfoSlot>
-      <p class="icon-text ml-7 pb-2 mt-8">
-        <v-icon color="success" class="pr-2">mdi-check</v-icon>
-        I have read, understood and agree to the terms and conditions of the Qualified Suppliers’ Agreement for the
-        Manufactured Home Registry.
-      </p>
+    <template v-if="getMhrUserAccessValidation.qsSaConfirmValid" v-slot:topInfoSlot>
```
Comment (orelbn) [inline ppr-ui/src/components/userAccess/ReviewConfirm/QsInformationReview.vue]: Can update to `#` to match other slot usage in file.

**Review by orelbn — APPROVED** [review]
Amazing!

File: ppr-ui/src/components/userAccess/ReviewConfirm/QsInformationReview.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/userAccess/ReviewConfirm/QsInformationReview.vue]: Good call, will do.
This was on your recommendation to shorthand, i think we all get it now (ie understand that they are slots) 👍

File: ppr-ui/src/views/userAccess/QsInformation.vue [hunk 2]
```diff
@@ -14,57 +14,92 @@
...
+          Read and accept the terms of the service agreement. You may download and a copy of this agreement to save for
+          your records.
+        </p>
 
-    <section class="qs-information-form mt-8">
-      <h2>Qualified Supplier ({{ getMhrSubProduct }}) Information</h2>
-      <p class="mt-1">
-        Provide the name and contact information for the Qualified Supplier.
-      </p>
+        <!-- download service agreement button -->
```
Comment (dimak1) [inline ppr-ui/src/views/userAccess/QsInformation.vue]: Would be a good candidate for common Download btn component

**Review by dimak1 — APPROVED** [review]
Looks good!

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1499-k6pag3qz.web.app

File: ppr-ui/src/views/userAccess/QsInformation.vue (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/views/userAccess/QsInformation.vue]: Sure, can do!

Comment (dimak1) [reply ppr-ui/src/views/userAccess/QsInformation.vue]: In the future, doesn't have to be in this PR, unless you have time.

Comment (cameron-eyds) [reply ppr-ui/src/views/userAccess/QsInformation.vue]: All good, already done!

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1499-k6pag3qz.web.app

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1499-k6pag3qz.web.app

File: ppr-ui/src/components/userAccess/ReviewConfirm/QsInformationReview.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/userAccess/ReviewConfirm/QsInformationReview.vue]: Updated

**Pull request merged**